transcom/mymove
# MBN-16222 Add handler for proof of service document upload

## [MB-16222](https://dp3.atlassian.net/browse/MB-16222)

## Summary

feat!: Add handler for uploading a service request document.
feat!: Add in new model for the service request documents on mto service items. Add in some model to payload and payload to model functions for the service request document as well.

I added a migration to drop a constraint on `service_request_documents_unique_key`, with the goal of this being able to upload multiple service request documents for a single service item.
The only difference (that was discernible to me) between how this operates and how the proof of service docs table operates is that the proof of service docs table also has `upload_id uuid NOT NULL`, which I couldn’t get working when I actually tried to upload documents for the service items.

I fixed the below error, leaving it here for posterity and reminder later if I have to do this again. 
Turns out, the thing that I needed from `primeAPI` wasn't called `ServiceRequestDocumentUploadHandler`, it was called `MtoServiceItemUpdateMTOServiceItemHandler`

I know part of it is waiting on the service object, but I'm also seeing an error in `api.go` for 
```go
primeAPI.ServiceRequestDocumentUploadHandler = CreateServiceRequestDocumentUploadHandler{
```

> primeAPI.ServiceRequestDocumentUploadHandler undefined (type *primeoperations.MymoveAPI has no field or method ServiceRequestDocumentUploadHandler)go


## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Access the office application.
2. Login as a a prime simulator user (may have to create a prime simulator user if there isn't one)
3. Access any of the moves.
4. Scroll down to the service items section. 
5. Click on "Upload Document for Test Service" 
6. Upload a document. 
7. A document should upload without issues. (And multiple documents should be able to be uploaded if you return to the same page to upload another document.) Note: The prime simulator doesn't allow uploading multiple documents in the one go, but if you return to the page again and upload another document, it should work.


### Backend



#### Any new migrations/schema changes:


[MB-16222]: https://dp3.atlassian.net/browse/MB-16222?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
